src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,123 @@
-    private final String[]      datePatterns;
+    private final Optional<TimeZone> tz;
+    private final String[]           datePatterns;
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
+    public JavaDateTimeHelper( Optional<TimeZone> tz, String... datePatterns ) {
         this.tz = tz;
         this.datePatterns = datePatterns;
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    private boolean shouldIgnoreValue( String date ) {
+        return date == null || StringUtils.isBlank( date ) || date.equals( "NULL" );
     }
 
-    public <R> R parseWithTwoDigitYearHandling(String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
+    public <R> R parseWithTwoDigitYearHandling(
+            String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
             BiFunction<R, String, R> postParseFunction ) {
-        if (shouldIgnoreValue(date)) {
+        if ( shouldIgnoreValue( date ) ) {
             return null;
         }
-        for ( String datePattern: datePatterns ){
+        for ( String datePattern : datePatterns ) {
             try {
                 DateTimeFormatter formatter = Cached.getDateFormatForString( datePattern );
                 R result = parseFunction.apply( date, formatter );
                 return postParseFunction.apply( result, datePattern );
-            } catch ( DateTimeParseException e) {
-                if ( datePattern.equals(datePatterns[datePatterns.length - 1]) ) {
-                    logger.error("Unable to parse date {}, please see debug log for additional information: {}.", date,e);
+            } catch ( DateTimeParseException e ) {
+                if ( datePattern.equals( datePatterns[ datePatterns.length - 1 ] ) ) {
+                    logger.error( "Unable to parse date {}, please see debug log for additional information: {}.",
+                            date,
+                            e );
                 }
             } catch ( ExecutionException ex ) {
-                logger.error("ExecutionException loading pattern from cache", ex);
+                logger.error( "ExecutionException loading pattern from cache", ex );
             }
+
         }
         return null;
     }
 
-    public LocalDate parseDate(String date) {
+    public LocalDate parseDate( String date ) {
         return parseWithTwoDigitYearHandling( date, LocalDate::parse, ( ld, datePattern ) -> {
-            if (  checkDatePatternIsTwoDigitYear( datePattern ) ) {
+            if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
                 // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((ld.getYear() - LocalDate.now().getYear()) > 20) {
-                    ld = ld.withYear(ld.getYear() - 100);
+                if ( ( ld.getYear() - LocalDate.now().getYear() ) > 20 ) {
```

Comment:
we should constantify these magic numbers

Reply:
done for 20, since that's variable.  100 is the era cut-off, so this is a natural constant.

Reply:
I wouldn't have known that 100 is the era cutoff, so I think it is still better to constantify it and call it `ERA_CUTOFF` or something so everyone looking at the code knows why it is being used quickly.

Reply:
➕